### PR TITLE
OJ 8375 - Convert Github adapter in the agent to use Normalized* dataclasses

### PR DESCRIPTION
**Pull request by tina-n219**

### CODE REVIEW

**Review by SomeKittens — APPROVED** [review]

File: jf_agent/git/__init__.py [hunk 1]
```diff
@@ -103,6 +103,13 @@ class NormalizedPullRequestComment:
     system_generated: bool = None
 
 
+@dataclass
+class NormalizedPullRequestRepository:
```
Comment (ametzger) [inline jf_agent/git/__init__.py:107]: Why not use [`NormalizedShortRepository`](https://github.com/jellyfish-ai/jf_agent/blob/3752f1a4f888a09d4b6bd18c859f4969c7115849/jf_agent/git/__init__.py#L63-L67)?

**Review by ametzger — CHANGES_REQUESTED** [review]
overall LGTM, one change

File: jf_agent/git/github.py [hunk 2]
```diff
@@ -289,63 +296,63 @@ def _get_merge_commit(client: GithubClient, pr, strip_text_content, redact_names
...
-        'deletions': pr['deletions'],
-        'changed_files': pr['changed_files'],
-        'comments': [
+        author=_normalize_user(client.get_json(pr['user']['url'])),
+        merged_by=(
+            _normalize_user(client.get_json(pr['merged_by']['url'])) if pr['merged'] else None
+        ),
+        commits=[
+            _normalize_commit(c, pr['base']['repo'], strip_text_content, redact_names_and_urls)
+            for c in tqdm(
```
Comment (daviskeene) [inline jf_agent/git/github.py:329]: Is TQDM necessary here?

File: jf_agent/git/github.py [hunk 3]
```diff
@@ -289,63 +296,63 @@ def _get_merge_commit(client: GithubClient, pr, strip_text_content, redact_names
...
+                client.get_pr_commits(pr['base']['repo']['full_name'], pr['number']),
+                f'downloading commits for PR {pr["number"]}',
+                leave=False,
+                unit='commits',
+            )
+        ],
+        merge_commit=_get_merge_commit(client, pr, strip_text_content, redact_names_and_urls),
+        comments=[
```
Comment (daviskeene) [inline jf_agent/git/github.py:337]: Should we go as far as to create dataclasses to model comments / approvals? Probably not necessary, just something I was thinking about while reviewing.

**Review by daviskeene — APPROVED** [review]
Looks good! Some comments/nits, but nothing outstanding that needs changing.

File: jf_agent/git/__init__.py (hunk 1 above)
Comment (tina-n219) [reply jf_agent/git/__init__.py:107]: whoops, missed that, good catch

File: jf_agent/git/github.py (hunk 3 above)
Comment (tina-n219) [reply jf_agent/git/github.py:337]: Good catch! I missed these, but we do have dataclasses to model comments and reviewws on pull requests

File: jf_agent/git/github.py (hunk 2 above)
Comment (tina-n219) [reply jf_agent/git/github.py:331]: good question, I can add that when this runs for orthog, the progress bars rarely show up as anything other than a quick blip. Not sure what it looks like for other companies

**Review by ametzger — APPROVED** [review]

**Pull request merged**